LearningMissionSimulation/PlaygroundMher.cs

```diff
@@ -4,59 +4,61 @@
 
 namespace LearningMissionSimulation
 {
-    public class PlaygroundMher
+    public class PlaygroundMher : ISimulation
     {
-        public PlaygroundMher()
-        {
-            
-        }
-       
-        //Account account = ObjectGenerator.GenerateAccount();
+        List<Account> studentList = new List<Account>();
+
+        List<Account> instructorList = new List<Account>();
 
         List<Account> accountList = new List<Account>();
 
-        Queue<Status> statusQueue = new Queue<Status>();
+        Queue<Account> accountQueueList = new Queue<Account>();
+
+        Dictionary<Guid, Account> accountDictionary = new Dictionary<Guid, Account>();
+
 
-        // Simulate account creation  process
         public void CreateAccounts(int accountCount)
         {
             int i = 0;
             while (i < accountCount)
             {
-                accountList.Add(ObjectGenerator.GenerateAccount());
-                i++;
-            }
-            i = 0;
-            foreach (var account in accountList)
-            {
-                if (account.Status == Status.Pending)
+                Account account = ObjectGenerator.GenerateAccount();
+
+                accountQueueList.Enqueue(account);
+
+                accountList.Add(account);
+
+                Console.WriteLine("========== Create  account {0} =======\n{1}",i, account);
+                Console.WriteLine("\n");
+
+                accountDictionary.Add(account.Id, account);
+                //I think it should be here switch
+                if (account.Role == Role.Instructor)
+                {
+                    instructorList.Add(account);
```

Comment:
Line 38 is incorrect. You have to generate an Instructor and add it to instructorList.